samples/nrf9160/azure_fota/src/main.c

```diff
@@ -244,8 +248,25 @@ void main(void)
 	modem_configure();
 	k_sem_take(&network_connected_sem, K_FOREVER);
 
-	static struct azure_iot_hub_config config = {
+#if IS_ENABLED(CONFIG_DEVICE_ID_USE_HW_ID)
+	char device_id[HW_ID_LEN];
+
+	err = hw_id_get(device_id, ARRAY_SIZE(device_id));
+	if (err) {
+		LOG_ERR("Failed to retrieve device ID");
+		return;
+	}
+#endif
+
+	struct azure_iot_hub_config config = {
 		.use_dps = IS_ENABLED(CONFIG_AZURE_IOT_HUB_DPS),
+#if IS_ENABLED(CONFIG_DEVICE_ID_USE_HW_ID)
+		.device_id = {
+			.ptr = device_id,
+			.size = strlen(device_id),
```

Comment:
```suggestion
			.size = IS_ENABLED(CONFIG_DEVICE_ID_USE_HW_ID) ? strlen(device_id)  : 0,
```

Might be a line width issue here. If it doesn't look good, just scrap it.